addons/point_of_sale/static/src/xml/Screens/PartnerListScreen/PartnerDetailsEdit.xml

```diff
@@ -13,50 +13,44 @@
+                               t-model="changes.street"
                                placeholder="Street" />
                     </div>
                     <div class="partner-detail">
                         <span class="label">City</span>
                         <input class="detail" name="city"
-                               t-on-change="captureChange" t-att-value="props.partner.city || ''"
+                               t-model="changes.city"
                                placeholder="City" />
                     </div>
                     <div class="partner-detail">
                         <span class="label">Postcode</span>
                         <input class="detail" name="zip"
-                               t-on-change="captureChange" t-att-value="props.partner.zip || ''"
+                               t-model="changes.zip"
                                placeholder="ZIP" />
                     </div>
-                    <div class="partner-detail">
+                    <div class="partner-detail" t-if="env.pos.states.some((state) => state.country_id[0] == changes.country_id)">
                         <span class="label">State</span>
-                        <select class="detail" name="state_id"
-                                t-on-change="captureChange">
+                        <select class="detail" name="state_id" t-model="changes.state_id">
                             <option value="">None</option>
                             <t t-foreach="env.pos.states" t-as="state" t-key="state.id">
-                                <option t-if="props.partner.country_id[0] == state.country_id[0]"
-                                        t-att-value="state.id"
-                                        t-att-selected="props.partner.state_id ? ((state.id === props.partner.state_id[0]) ? true : undefined) : undefined">
+                                <option t-if="changes.country_id == state.country_id[0]" t-att-value="state.id">
                                     <t t-esc="state.name" />
                                 </option>
                             </t>
                         </select>
                     </div>
                     <div class="partner-detail">
                         <span class="label">Country</span>
-                        <select class="detail" name="country_id"
-                                t-on-change="captureChange">
+                        <select class="detail" name="country_id" t-model="changes.country_id">
                             <option value="">None</option>
                             <t t-foreach="env.pos.countries" t-as="country" t-key="country.id">
-                                <option t-att-value="country.id"
-                                        t-att-selected="props.partner.country_id ? ((country.id === props.partner.country_id[0]) ? true : undefined) : undefined">
```

Comment:
The "selected" attribute is needed so that the "select" element knows which to show when the options are hidden. Make sure to take this fix (https://github.com/odoo/odoo/pull/127581) into account.